src/App.jsx

```diff
@@ -14,33 +16,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreperedGoods(goods, sortField, toReverse) {
+  let preperedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABET:
+        preperedGoods = preperedGoods.sort((a, b) => a.localeCompare(b));
+        break;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      case SORT_FIELD_LENGTH:
+        preperedGoods = preperedGoods.sort((a, b) => a.length - b.length);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  if (toReverse) {
+    preperedGoods.reverse();
+  }
+
+  return preperedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [toReverse, setToReverse] = useState(false);
+  const visibleGoods = getPreperedGoods(goodsFromServer, sortField, toReverse);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SORT_FIELD_ALPHABET,
+          })}
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !toReverse,
+          })}
+          onClick={() => (toReverse ? setToReverse(false) : setToReverse(true))}
```

Comment:
'

```suggestion
          onClick={() => (setIsReversed(!isReversed)}
```
looks strange